Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -106,10 +155,16 @@ public override void ChangeDatabase(string databaseName)
         public override void Close()
         {
             logger.Debug("Close Connection.");
-            if ((_connectionState != ConnectionState.Closed) &&
-                (SfSession != null))
+            if (IsNonClosedWithSession())
             {
-                if (SnowflakeDbConnectionPool.addSession(SfSession))
+                var transactionInProgress = GetTransactionId();
+                if (transactionInProgress != DBNull.Value)
+                {
+                    if (TerminateTransaction(transactionInProgress))

```

Comment:
Ok, I remember our conversation about that topic. But what do you think about throwing exception after rolling back if someone forgot to end transaction? Fail fast... We can discuss it today.

Reply:
Check my reply to @sfc-gh-dprzybysz comments

Reply:
Ok, fair enough.